### ISSUE TITLE
HTTPS downloads fail with Turi Create 5.7

### ISSUE THREAD

**Issue by syoutsey** [apple/turicreate#2384]
HTTPS downloads fail with Turi Create 5.7. This issue does not reproduce on Turi Create 5.6.

System: macOS 10.15 beta

Error: `OSError: Cannot open https://static.turi.com/datasets/audioscrobbler for read. Not implemented: compiled without support for http(s):// URLs.`

Reproduction steps - working version:

```
(venv) ➜  turi-create-3.6-5.6 pip list | grep turicreate
turicreate      5.6        
(venv) ➜  turi-create-3.6-5.6 python --version
Python 3.6.8
(venv) ➜  turi-create-3.6-5.6 python
Python 3.6.8 (v3.6.8:3c6b436a57, Dec 24 2018, 02:04:31) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import turicreate as tc
>>> sf = tc.SFrame('https://static.turi.com/datasets/audioscrobbler')
Downloading https://static.turi.com/datasets/audioscrobbler/dir_archive.ini to /var/folders/bt/d7_f7y897cz8f7256sj30j1c0000gn/T/turicreate-sam/41175/2c04d8ef-585a-4937-89f5-3c8ba78d52b9.ini
Downloading https://static.turi.com/datasets/audioscrobbler/objects.bin to /var/folders/bt/d7_f7y897cz8f7256sj30j1c0000gn/T/turicreate-sam/41175/1f709ccd-edfe-4cc0-9c4c-175ef6f38ef4.bin
Downloading https://static.turi.com/datasets/audioscrobbler/m_74b0dc51.frame_idx to /var/folders/bt/d7_f7y897cz8f7256sj30j1c0000gn/T/turicreate-sam/41175/aa6f3d62-9b01-4071-8d8c-db5291cf3276.frame_idx
Downloading https://static.turi.com/datasets/audioscrobbler/m_74b0dc51.sidx to /var/folders/bt/d7_f7y897cz8f7256sj30j1c0000gn/T/turicreate-sam/41175/561e4baa-7d91-4a81-a5a4-24d5785c5a9d.sidx
```

Reproduction steps - nonfunctional version:

```
(venv) ➜  turi-create-3.6-5.7 pip list | grep turicreate                                                    
turicreate      5.7        
(venv) ➜  turi-create-3.6-5.7 python --version
Python 3.6.8
(venv) ➜  turi-create-3.6-5.7 python
Python 3.6.8 (v3.6.8:3c6b436a57, Dec 24 2018, 02:04:31) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import turicreate as tc
sf = tc.SFrame('https://static.turi.com/datasets/audioscrobbler')
>>> sf = tc.SFrame('https://static.turi.com/datasets/audioscrobbler')
Downloading https://static.turi.com/datasets/audioscrobbler to /var/folders/bt/d7_f7y897cz8f7256sj30j1c0000gn/T/turicreate-sam/42755/8ae0d96b-935c-4ecd-a0a5-f6fa33b9bc83
Failed to download https://static.turi.com/datasets/audioscrobbler: HTTP response code said error
Traceback (most recent call last):
  File "/Users/sam/src/explore/turi-create-3.6-5.7/venv/lib/python3.6/site-packages/turicreate/data_structures/sframe.py", line 808, in __init__
    self.__proxy__.load_from_sframe_index(url)
  File "cy_sframe.pyx", line 71, in turicreate._cython.cy_sframe.UnitySFrameProxy.load_from_sframe_index
  File "cy_sframe.pyx", line 74, in turicreate._cython.cy_sframe.UnitySFrameProxy.load_from_sframe_index
OSError: Cannot open https://static.turi.com/datasets/audioscrobbler for read. Not implemented: compiled without support for http(s):// URLs.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/sam/src/explore/turi-create-3.6-5.7/venv/lib/python3.6/site-packages/turicreate/data_structures/sframe.py", line 812, in __init__
    raise ValueError('Unknown input type: ' + format)
  File "/Users/sam/src/explore/turi-create-3.6-5.7/venv/lib/python3.6/site-packages/turicreate/_cython/context.py", line 49, in __exit__
    raise exc_type(exc_value)
OSError: Cannot open https://static.turi.com/datasets/audioscrobbler for read. Not implemented: compiled without support for http(s):// URLs.
>>> 
```

**Comment by hoytak** [comment]
This could intersect with https://github.com/apple/turicreate/pull/2387.  This fixes a few places where the if-defs were wrong.

**Comment by hoytak** [comment]
Could we confirm that this works with 5.6?  In the browser it says it's invalid.

**Comment by shantanuchhabra** [comment]
Yes, on it right now.

**Comment by znation** [comment]
It's valid if you assume it's a directory filled with SFrame contents.

**Comment by hoytak** [comment]
Ah, I see. Yes, then this is a regression.

**Comment by Jarvi-Izana** [comment]
Do we have a sample http/https dataset we can use to do integration test?

**Comment by syoutsey** [comment]
@Jarvi-Izana I'm not sure what you mean, does the SFrame linked in the original error report work?

**Comment by Jarvi-Izana** [comment]
@syoutsey Yes, it's fixed in #2405. I need another link to a small sample data for test purpose to make sure we won't break reading from https/http in the future.

**Comment by syoutsey** [comment]
@Jarvi-Izana does this work? https://static.turi.com/datasets/nips-text

**Comment by Jarvi-Izana** [comment]
@syoutsey That doesn't contain frame formatted data. Do you have dataset similar to `https://static.turi.com/datasets/audioscrobbler` but with small size, probably 1 row is enough for us to test pull data over real network.

**Comment by Jarvi-Izana** [comment]
fixed by #2405
